docs/collection/CMakeLists.txt

```diff
@@ -6,8 +6,9 @@
 #
 # Contact KDAB at <info@kdab.com> for commercial licensing options.
 #
-if(NOT TARGET Qt5::qcollectiongenerator)
-    message(STATUS "qcollectiongenerator not found, documentation collection will not be generated.")
+find_package(Qt${Qt_VERSION_MAJOR} NO_MODULE QUIET OPTIONAL_COMPONENTS Help ToolsTools)
```

Comment:
I think ToolsTools is not available for Qt5.
could you move `find_package(Qt${Qt_VERSION_MAJOR} NO_MODULE QUIET OPTIONAL_COMPONENT ToolsTools)` inside a conditional for !Qt5 or somesuch?

Reply:
@winterz Correct, it's not available for Qt5. And `AttributionsScannerTools` isn't available in Qt6. That's why these are all optional components.

This `find_package()` will import the CMake targets for `Qt::qdoc` and friends for either Qt5 or Qt6, even though those targets are defined in **different places** in the two versions. The ones that match the version in use will be loaded, the others will be unavailable.

Reply:
Anyway, yes, they can be split up into separate branches if you'd prefer, sure. There's no _harm_ in trying to import Qt5ToolsTools that will never exist, but it also isn't an imperative that we try to import it under Qt5.

Though, with that other branch I mentioned, I was actually planning to expand this with _additional_ (optional) imports to discover `qdoc` and `qattributionsscanner` using CMake targets as well (when possible), and some of those will not only not-exist in one of Qt5 or Qt6, but whether they exist will depend on the _version_ of Qt5 in use. Versions earlier than ~Qt 5.13 will be missing some CMake configs that _will_ exist in later versions, so there will still be potentially-nonexistent optional imports in the Qt5 branch. (The code would fall back to the current `find_program()`-based discovery, in those cases. But only as a fallback if the targets can't be imported from the optional components.)

However, `Qt5ToolsTools` will never exist under any circumstance, it's true.